autoscheduler/frontend/src/components/SchedulingPage/Schedule/MeetingCard/MeetingCard.tsx

```diff
@@ -3,32 +3,157 @@ import { Typography } from '@material-ui/core';
 import * as styles from './MeetingCard.css';
 import Meeting, { MeetingType } from '../../../../types/Meeting';
 import ScheduleCard from '../ScheduleCard/ScheduleCard';
+import meetingTimeText from '../../../../utils/meetingTimeText';
+import { meetingBuildingWithRoom } from '../../../../utils/meetingBuilding';
+
+enum MeetingCardSizes {
+  small, // 1 line, for <50 minute meetings
+  medium, // 2 lines
+  large // 3 lines
+}
 
 interface MeetingCardProps {
   meeting: Meeting;
   bgColor: string;
   firstHour: number;
   lastHour: number;
+  fullscreen?: boolean;
 }
 
 /**
  * Renders a meeting of a class on the schedule
  * @param props include meeting, bgColor, firstHour, and lastHour
  */
 const MeetingCard: React.FC<MeetingCardProps> = ({
-  meeting, bgColor, firstHour, lastHour,
+  meeting, bgColor, firstHour, lastHour, fullscreen = false,
 }: MeetingCardProps) => {
   // destructure meeting for ease of access
   const {
     startTimeHours, startTimeMinutes, endTimeHours, endTimeMinutes, section, meetingType,
   } = meeting;
-  const [isBig, setIsBig] = React.useState(true);
+  const [cardSize, setCardSize] = React.useState(MeetingCardSizes.large);
 
-  // hide meeting type if the card is small
-  const handleResize = React.useCallback((newVal: boolean): void => {
-    setIsBig(newVal);
+  // Determines what size the card is, depending on the difference between the desired and available
+  // height.
+  const handleResize = React.useCallback((contentHeight: number, clientHeight: number): void => {
+    const heightDiff = clientHeight - contentHeight;
+    if (heightDiff > 30) {
+      setCardSize(MeetingCardSizes.large);
+    } else if (heightDiff > -15) { // Just enough for two rows of text
+      setCardSize(MeetingCardSizes.medium);
+    } else {
+      setCardSize(MeetingCardSizes.small);
+    }
   }, []);
 
+  const fullscreenLargeCard = (
```

Comment:
Should I declare these in the switch so it doesn't construct these until we actually need them, since we're making 3 always but only ever need 1 per card? Or does it not matter since it's not rendered/executed until we actually render them

Reply:
Yeah do it in the switch